libraries/tickv/src/async_ops.rs

```diff
@@ -541,26 +549,36 @@ mod tests {
                 }
 
                 self.run.set(self.run.get() + 1);
-
-                Ok(())
+                self.waiting_on.set(FlashCtrlAction::Write);
+                Err(ErrorCode::WriteNotReady(address))
             }
 
             fn erase_region(&self, region_number: usize) -> Result<(), ErrorCode> {
                 println!("Erase region: {}", region_number);
 
-                if self.async_erase_region.get() != region_number {
-                    // Pretend that we aren't ready
-                    self.async_erase_region.set(region_number);
-                    return Err(ErrorCode::EraseNotReady(region_number));
-                }
-
                 let mut local_buf = self.buf.borrow_mut()[region_number];
 
                 for d in local_buf.iter_mut() {
                     *d = 0xFF;
                 }
 
-                Ok(())
+                // Pretend that we aren't ready
+                self.async_erase_region.set(region_number);
+
+                self.waiting_on.set(FlashCtrlAction::Erase);
+                Err(ErrorCode::EraseNotReady(region_number))
+            }
+        }
+
+        fn flash_ctrl_callback(tickv: &AsyncTicKV<FlashCtrl, 1024>) {
```

Comment:
Can we document this function so it's obvious what it is doing?

Reply:
Done.